Gems/Atom/RPI/Code/Source/RPI.Public/Material/Material.cpp

```diff
@@ -600,13 +600,23 @@ namespace AZ
             }
             else
             {
+                AZ::Data::AssetInfo assetInfo;
+                AZ::Data::AssetCatalogRequestBus::BroadcastResult(assetInfo, &AZ::Data::AssetCatalogRequests::GetAssetInfoById, imageAsset.GetId());
+
+                auto assetType = imageAsset.GetType();
+
+                if (assetInfo.m_assetId.IsValid())
+                {
+                    assetType = assetInfo.m_assetType;
+                }
+
```

Comment:
If you're making the call to get the asset info, can you remove the check and assume the one from the asset info is always correct?

Reply:
I think for the assets created at runtime, it won't have assetinfo.

Reply:
On the other hand, why is GetAssetInfoById needed? Why isn't the type info in the Asset<> correct?
(This probably deserves a code comment).

Reply:
It's kind of fail safe check to correct it if any input went wrong. 
I hope I correct every single places which construct the data<asset> but I can't promise. This would keep things still function.